### PR TITLE
[CDAP-8441] Make ConfigurationCheck more robust and informative

### PR DESCRIPTION
**Pull request by anew**
now ConfigurationCheck catches all possible kinds of exceptions. Also little better, unified error message across the board.

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/master/startup/ServiceResourceKeys.java [hunk 1]
```diff
@@ -74,4 +74,8 @@ public String getMaxInstancesKey() {
   public int getMaxInstances() {
     return cConf.getInt(maxInstancesKey);
   }
+
+  public String get(String key) {
```
Comment (chtyim) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/ServiceResourceKeys.java]: Is this method used? I don't see it in this PR.

File: cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java [hunk 2]
```diff
@@ -130,73 +133,103 @@ private void checkPotentialPortConflicts() {
...
   }
 
+  /**
+   * Best effort to log a user-friendly error message for a badly configured key and value.
+   *
+   * @param message The message desxcribing the problem
+   * @param e The exception that caused the problem.
+   */
+  private void logProblem(String message, String key, String value, Exception e) {
+    if (e == null || value == null) {
```
Comment (chtyim) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java]: Adding `@Nullable` to `value` and `e`?

File: cdap-master/src/main/java/co/cask/cdap/master/startup/ServiceResourceKeys.java (hunk 1 above)
Comment (anew) [reply cdap-master/src/main/java/co/cask/cdap/master/startup/ServiceResourceKeys.java]: Oh I first added it and then anded up not needing it... removed.

File: cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java (hunk 2 above)
Comment (anew) [reply cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java]: Good point. Done

**Comment by chtyim** [comment]
LGTM

**Comment by anew** [comment]
Previous build passed and I only removed an unused method. Squashing and merging now.

**Pull request merged**